ui/admin/app/controllers/scopes/scope/session-recordings/index.js

```diff
@@ -95,29 +106,95 @@ export default class ScopesScopeSessionRecordingsIndexController extends Control
   }
 
   /**
-   * Returns unique project scopes from targets
-   * linked to the session recordings
+   * Configurations for different filter options
    * @returns {[object]}
    */
-  get projectScopes() {
-    const uniqueMap = new Map();
-    this.model.allSessionRecordings.forEach(
-      ({
-        create_time_values: {
-          target: {
-            scope: { id, name, parent_scope_id },
-          },
+  get filterConfigs() {
+    return {
+      userFilters: {
+        select: [
+          { field: 'user_id', isDistinct: true },
+          { field: 'user_name', isDistinct: true },
+        ],
+        searchFields: ['user_id', 'user_name'],
+        mapper: ({ user_id, user_name }) => ({
+          id: user_id,
+          name: user_name,
+        }),
+      },
+      scopeFilters: {
+        select: [
+          { field: 'target_scope_id', isDistinct: true },
+          { field: 'target_scope_name', isDistinct: true },
+          { field: 'target_scope_parent_scope_id', isDistinct: true },
+        ],
+        searchFields: [
+          'target_scope_id',
+          'target_scope_name',
+          'target_scope_parent_scope_id',
+        ],
+        mapper: ({
+          target_scope_id,
+          target_scope_name,
+          target_scope_parent_scope_id,
+        }) => ({
+          id: target_scope_id,
+          name: target_scope_name ?? target_scope_id,
+          parent_scope_id: target_scope_parent_scope_id,
+        }),
+      },
+      targetFilters: {
+        select: [
+          { field: 'target_id', isDistinct: true },
+          { field: 'target_name', isDistinct: true },
+        ],
+        searchFields: ['target_id', 'target_name'],
+        mapper: ({ target_id, target_name }) => ({
+          id: target_id,
+          name: target_name,
+        }),
+      },
+    };
+  }
+
+  /**
+   * Generic retrieve function for session recording options
+   * @param {string} type - The type of options to retrieve (user, scope, target)
+   * @param {string} search - Search term
+   * @returns {Promise<Array>}
+   */
+  async retrieveFilterOptions(type, search) {
+    const config = this.filterConfigs[type];
+    assert(`Unknown filter type: ${type}`, config);
+
+    const options = {
+      peekDb: true,
+      returnRawData: true,
+    };
+    const results = await this.store.query(
+      'session-recording',
+      {
+        select: config.select,
+        query: {
+          search: { text: search, fields: config.searchFields },
         },
-      }) => {
-        if (!uniqueMap.has(id)) {
-          const projectName = name || id;
-          uniqueMap.set(id, { id, name: projectName, parent_scope_id });
-        }
+        page: 1,
+        pageSize: 250,
```

Comment:
Should we also update the Dropdown component item limit from 500 to 250 to keep it consistent (since its used in desktop app too)?